src/plugins/ucc_discovery_plugin.cpp

```diff
@@ -110,7 +110,7 @@ void UccDiscoveryPlugin::_validate_ucc_candidates(const UccCandidates& ucc_candi
     if (std::any_of(soft_key_constraints.cbegin(), soft_key_constraints.cend(),
                     [&column_id](const auto& key_constraint) {
                       const auto& columns = key_constraint.columns();
-                      return columns.size() == 1 && columns.front() == column_id;
+                      return columns.size() == 1 && *columns.cbegin() == column_id;
```

Comment:
why?

Reply:
`.front()` vs. `*.cbegin()`? We discussed to remove `columns` as a member of the `AbstractTableConstraint`. Since a key constraint has set semantics, it now holds a set of columns – and a set has no `.front()`

Reply:
Got it. Wasn't aware a sorted set does not have `front()`.